src/prefect/agent/agent.py

```diff
@@ -356,7 +356,8 @@ def _deploy_flow_run(
             # Wait for the flow run's start time. The agent pre-fetches runs that may
             # not need to start until up to 10 seconds later so we need to wait to
             # prevent the flow from starting early
-            start_time = pendulum.parse(flow_run.scheduled_start_time)
+            flow_run_state = StateSchema().load(flow_run.serialized_state)
+            start_time = getattr(flow_run_state, "start_time", pendulum.now())
```

Comment:
There should be a comment explaining why we are using the state start time and not the `scheduled_start_time`.

Also a comment explaining that we may be working with both `Running` and `Scheduled` states and that's why we need to use `getattr`